STYLEGUIDE.adoc

```diff
@@ -18,14 +18,66 @@ The contents of this document are “recommendations”, that is:
 
 === Basic formatting
 
-Use 4 spaces for indentation. Avoid lines over 100 characters. This is codified in `.editorconfig`.
+Use 2 spaces for indentation. Full-indent for `where` is recommended.
+
+Bad:
+
+[source,Haskell]
+----
+f x = y
+ where
+  y = ...
+----
+
+Good:
+
+[source,Haskell]
+----
+f x = y
+  where
+    y = ...
+----
+
+Avoid lines over 100 characters. This is codified in `.editorconfig`.
 
 Format Haskell source with `stylish-haskell` (config in `.stylish-haskell.yaml`).
+This is enforced in the pre-commit hook and in CI.
+Formatting with `stylish-haskell` must be a no-op.
+
+Type signatures should have the double-colon on the same line as the name, separated by one space.
+
+Bad:
+
+[source,Haskell]
+----
+foo
+  :: A -> B
+  -> C
+----
+
+Good:
+
+[source,Haskell]
+----
+foo :: A -> B -> C
+----
+
+Good:
+[source,Haskell]
+----
+foo ::
```

Comment:
Example with constraints and explicit quantification, please

So the thing I think we actually currently do much more (and the one I prefer) is:
```
foo 
   :: forall a b c
   . (C1 a
   , C2 b)
   => A
   -> B
   -> C
```

Reply:
Added. I still prefer to have `::` on the first line, so

```haskell
foo ::
  forall a b c.
  ( C1 a
  , C2 b
  ) =>
  A ->
  B ->
  C
```

This is also fine:

```haskell
foo ::
  forall a b c.
  (C1 a, C2 b) =>
  A -> B -> C
```